core/space/services/services_sharing.go

```diff
@@ -225,12 +225,30 @@ func (s *Space) OpenSharedFile(ctx context.Context, hash, password, filename str
 	}, nil
 }
 
-func (s *Space) ShareFilesViaPublicKey(ctx context.Context, bucketName string, paths []string, pubkeys []crypto.PubKey) error {
-	err := s.tc.ShareFilesViaPublicKey(ctx, bucketName, paths, pubkeys)
+func (s *Space) ShareFilesViaPublicKey(ctx context.Context, paths []domain.FullPath, pubkeys []crypto.PubKey) error {
+	err := s.tc.ShareFilesViaPublicKey(ctx, paths, pubkeys)
 	if err != nil {
 		return err
 	}
+
+	for _, path := range paths {
+		if path.DbId == "" {
+			// TODO: fetch dbid of mirror bucket and set
+			// b, err := s.tc.GetDefaultBucket(ctx)
+			// if err != nil {
+			// 	return err
+			// }
+
+			// bs, err := s.tc.GetBucketSchema(ctx, b.Slug())
+			// if err != nil {
+			// 	return err
+			// }
+			// path.DbId == bs.dbId
+		}
+	}
+
 	for _, pk := range pubkeys {
+
 		d := &domain.Invitation{
 			Paths: paths,
 			// Key: TODO - get from keys thread for each file
```

Comment:
@perfectmak could you direct me to somewhere to be able to fetch the keys given a path? I can create one if there isn't one coming with your work.

Reply:
I added the encryption key to the bucketschema here: https://github.com/FleekHQ/space-daemon/pull/123/files#diff-8ea426b2ef49623ad0f07dd826f93169R21

It is bucket level and works for all paths under the parent bucket.